src/v/cluster/metadata_dissemination_handler.cc

```diff
@@ -21,12 +21,32 @@
 #include "rpc/connection_cache.h"
```

Comment:
I totally agree here with @dotnwat. We do not have to disseminate leadership information for materialized partitions since information about source partition leadership is already available. It is enough to ask about source partition leadership.